Content.Shared/Chemistry/Components/HyposprayComponent.cs

```diff
@@ -0,0 +1,49 @@
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+using Robust.Shared.Audio;
+
+namespace Content.Shared.Chemistry.Components;
+
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class HyposprayComponent : Component
+{
+    [DataField]
+    public string SolutionName = "hypospray";
+
+    // TODO: This should be on clumsycomponent.
+    [DataField]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float ClumsyFailChance = 0.5f;
+
+    [DataField]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 TransferAmount = FixedPoint2.New(5);
+
+    [DataField]
+    public SoundSpecifier InjectSound = new SoundPathSpecifier("/Audio/Items/hypospray.ogg");
+
+    /// <summary>
+    /// Whether or not the hypo is able to inject only into mobs. On false you can inject into beakers/jugs
+    /// </summary>
+    [AutoNetworkedField]
+    [DataField]
```

Comment:
You should probably make this required to match the previous implementation. It's also generally a good idea to make properties that influence functionality required because otherwise the user will make assumptions about what the default case is (this leads to bugs)

Reply:
How would I make it required?

Reply:
Okay I figured it out, now that I look at it again, I realize this doesn't need the DataField tag at all.
Setting the initial mode of the Hypospray isn't something that people need to really set in the YML, since they can just toggle between the two at any point. Am I find to just remove the DataField attribute?

Reply:
I have been informed on Discord that DataFields are necessary for stateful attributes to load properly during replays so.... back it goes!

Reply:
> I have been informed on Discord that DataFields are necessary for stateful attributes to load properly during replays so.... back it goes!

This isn't correct with how replays are implemented. That said it definitely still should be a data field regardless.